src/TesApi.Web/BatchScheduler.cs

```diff
@@ -1858,10 +1858,16 @@ public async ValueTask<bool> ActionAsync(TesTask tesTask, CombinedBatchTaskInfo
 
         private class ExternalStorageContainerInfo
         {
+            private string sasToken;
+
             public string AccountName { get; set; }
             public string ContainerName { get; set; }
             public string BlobEndpoint { get; set; }
-            public string SasToken { get; set; }
+            public string SasToken
+            {
+                get => sasToken;
+                set => sasToken = value.TrimStart('?');
```

Comment:
add the null-conditional operator in case someone tries to set as null

`set => sasToken = value?.TrimStart('?');`